server/app/services/auth-service.js

```diff
@@ -432,6 +432,51 @@ async function usernameVerify(userId, username, password, usernameToken) {
   await transaction.commit();
 }
 
+/**
+ * Cancel a username update request.
+ *
+ * @param userId {Number} user id
+ * @param username {String} User's login
+ * @returns {Promise<void>} Nothing
+ */
+async function cancelUsernameUpdate(userId, email) {
+  const co = await ConnectionInformation.findById(userId);
+
+  if (!co) {
+    throw createUserError('VerificationError', 'No user found for this user id.');
+  }
+
+  const transaction = await sequelize.transaction();
+
+  try {
+    await co.update({
+      usernameToken: null,
+    }, { transaction });
+
+    await JWT.update({ revoked: true }, {
```

Comment:
Do we need to revoke every token when someone cancel his email change ?

Reply:
You are right ! :)